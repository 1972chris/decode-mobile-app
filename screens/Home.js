import React from 'react';
import { connect } from 'react-redux';
import { Image, Text, TextInput, View, KeyboardAvoidingView, Keyboard } from 'react-native';
import { SecureStore, ScreenOrientation } from 'expo';
import PropTypes from 'prop-types';
import { goToAttributesLanding, goToAttributesSummary, goToPetitionSummary } from '../application/redux/actions/navigation';
import { onStartApp } from '../application/redux/actions/petitionLink';
import { loadCredentials } from '../application/redux/actions/attributes';
import { getWalletId } from '../application/redux/actions/wallet';
import authorizationAction, { updatePin } from '../application/redux/actions/authorization';
import { storePinOnAppInitalization } from '../LocalStorage';
import Button from '../application/components/Button/Button';


import styles from './styles';

const decodeLogo = require('../assets/images/decode-hexagon.png');

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.validatePinCode = this.validatePinCode.bind(this);
  }

  componentWillMount() {
    ScreenOrientation.allow(ScreenOrientation.Orientation.PORTRAIT_UP);
    this.props.initializeState().then(() => {});
  }

  goToPetition() {
    const isAttributeVerified = this.props.attributes.nonVerified.length > 0;
    if (isAttributeVerified) {
      this.props.goToPetitionSummary(this.props.petitionLink);
    } else {
      this.props.goToAttributesSummary(this.props.petitionLink);
    }
  }

  goToNextPage() {
    if (this.props.petitionLink) {
      this.goToPetition();
    } else {
      this.props.goToAttributesLanding();
    }
  }

  validatePinCode() {
    return this.props.doAuthorize(this.props.pinCode).then((action) => {
      if (action.pinCorrect) {
        Keyboard.dismiss();
        this.goToNextPage();
      } else {
        alert('Incorrect pin code'); // eslint-disable-line
      }
    });
  }

  render() {
    return (
      <KeyboardAvoidingView
        behavior="position"
        keyboardVerticalOffset={50}
      >
        <View style={styles.homeContainer}>
          <Image
            style={styles.homeLogo}
            source={decodeLogo}
          />
          <View style={styles.homeTextInput}>
            <Text style={{ marginVertical: 10 }}>PIN:&nbsp;&nbsp;</Text>
            <TextInput
              style={styles.homePassword}
              placeholder=" Pin"
              keyboardType="numeric"
              secureTextEntry
              underlineColorAndroid="transparent"
              onChangeText={pin => this.props.updatePin(pin)}
              value={this.props.pinCode}
            />
          </View>
          <View style={{ flexDirection: 'row' }}>
            <Button
              name="Sign In"
              onPress={this.validatePinCode}
              style={{
                width: 150,
                alignItems: 'center',
              }}
            />
          </View>
        </View>
      </KeyboardAvoidingView>
    );
  }
}

Home.propTypes = {
  goToAttributesLanding: PropTypes.func.isRequired,
  goToAttributesSummary: PropTypes.func.isRequired,
  goToPetitionSummary: PropTypes.func.isRequired,
  initializeState: PropTypes.func.isRequired,
  doAuthorize: PropTypes.func.isRequired,
  updatePin: PropTypes.func.isRequired,
  petitionLink: PropTypes.string,
  pinCode: PropTypes.string,
  attributes: PropTypes.shape({
    nonVerified: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
};

Home.defaultProps = {
  petitionLink: undefined,
  pinCode: '',
};

const mapStateToProps = state => ({
  petitionLink: state.petitionLink.petitionLink,
  pinCode: state.authorization.pin,
  attributes: state.attributes,
});

const mapDispatchToProps = dispatch => ({
  goToAttributesLanding: () => { dispatch(goToAttributesLanding()); },
  goToAttributesSummary: (petitionLink) => { dispatch(goToAttributesSummary(petitionLink)); },
  goToPetitionSummary: (petitionLink) => { dispatch(goToPetitionSummary(petitionLink)); },
  doAuthorize: pin => dispatch(authorizationAction(pin, SecureStore.getItemAsync)),
  updatePin: (pin) => { dispatch(updatePin(pin)); },
  initializeState: async () => {
    await dispatch(onStartApp());
    await dispatch(getWalletId());
    await storePinOnAppInitalization(SecureStore.setItemAsync);
    await dispatch(loadCredentials(SecureStore.getItemAsync));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);

