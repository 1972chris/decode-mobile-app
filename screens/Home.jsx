import React from 'react';
import { StyleSheet, Image, Text, TextInput, View, Button } from 'react-native';

const decodeLogo = require('../assets/images/decode_logo.png');

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#fff',
    flex: 1,
    paddingTop: 100,
  },
  logo: {
    height: 100,
    marginBottom: 20,
    width: 200,
  },
  text: {
    fontFamily: 'Montserrat-Medium',
    marginBottom: 40,
    marginTop: 40,
  },
});

function Home() {
  return (
    <View style={styles.container}>
      <Image
        style={styles.logo}
        source={decodeLogo}
      />
      <Text style={styles.text}>Welcome to the DECODE Wallet</Text>
      <TextInput
        placeholder="Password"
        style={{ height: 40, width: 250, marginBottom: 20 }}
        underlineColorAndroid="rgb(0,163,158)"
      />
      <Button
        onPress={() => true}
        title="Unlock"
        color="rgb(0,163,158)"
      />
    </View >
  );
}

export default Home;
