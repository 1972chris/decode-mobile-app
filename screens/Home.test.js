import React from 'react';
import { expect } from 'chai';
import renderer from 'react-test-renderer';
import Home from './Home';

it('renders Home component', () => {
  const rendered = renderer.create(<Home />).toJSON();
  expect(rendered).to.exist;
});
