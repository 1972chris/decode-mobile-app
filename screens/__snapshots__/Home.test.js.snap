// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    goQRScannerIntro={[Function]}
    goToAuthorization={[Function]}
    loadWalletId={[Function]}
    petitionLink={undefined}
    petitionLinkStartup={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={1}
          style={
            Object {
              "height": 150,
              "marginBottom": 50,
              "marginTop": 75,
              "width": 300,
            }
          }
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
              "marginBottom": 30,
            }
          }
        >
          Welcome, Jane Doe
        </Text>,
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "rgb(0,163,158)",
              "marginBottom": 20,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            placeholder="Password"
            secureTextEntry={true}
            style={
              Object {
                "height": 30,
                "width": 300,
              }
            }
            underlineColorAndroid="rgb(0,163,158)"
          />
        </View>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "rgb(0,163,158)",
              "borderRadius": 2,
              "elevation": 2,
              "height": 36,
              "marginBottom": 8,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "shadowColor": "rgba(0, 0, 0, 0.54)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
              "width": 300,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFF",
                "fontSize": 14,
                "fontWeight": "500",
              }
            }
          >
            LOG IN
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Object {
            "height": 150,
            "marginBottom": 50,
            "marginTop": 75,
            "width": 300,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Welcome, Jane Doe",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 18,
            "marginBottom": 30,
          },
        },
        "ref": null,
        "rendered": "Welcome, Jane Doe",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextInput
            allowFontScaling={true}
            placeholder="Password"
            secureTextEntry={true}
            style={
              Object {
                "height": 30,
                "width": 300,
              }
            }
            underlineColorAndroid="rgb(0,163,158)"
          />,
          "style": Object {
            "borderBottomWidth": 1,
            "borderColor": "rgb(0,163,158)",
            "marginBottom": 20,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "placeholder": "Password",
            "secureTextEntry": true,
            "style": Object {
              "height": 30,
              "width": 300,
            },
            "underlineColorAndroid": "rgb(0,163,158)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFF",
                "fontSize": 14,
                "fontWeight": "500",
              }
            }
          >
            LOG IN
          </Text>,
          "onPress": [Function],
          "style": Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,163,158)",
            "borderRadius": 2,
            "elevation": 2,
            "height": 36,
            "marginBottom": 8,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "shadowColor": "rgba(0, 0, 0, 0.54)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 2,
            "width": 300,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "LOG IN",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "color": "#FFF",
              "fontSize": 14,
              "fontWeight": "500",
            },
          },
          "ref": null,
          "rendered": "LOG IN",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "height": 150,
                "marginBottom": 50,
                "marginTop": 75,
                "width": 300,
              }
            }
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "marginBottom": 30,
              }
            }
          >
            Welcome, Jane Doe
          </Text>,
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "rgb(0,163,158)",
                "marginBottom": 20,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="Password"
              secureTextEntry={true}
              style={
                Object {
                  "height": 30,
                  "width": 300,
                }
              }
              underlineColorAndroid="rgb(0,163,158)"
            />
          </View>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "rgb(0,163,158)",
                "borderRadius": 2,
                "elevation": 2,
                "height": 36,
                "marginBottom": 8,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "shadowColor": "rgba(0, 0, 0, 0.54)",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.8,
                "shadowRadius": 2,
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#FFF",
                  "fontSize": 14,
                  "fontWeight": "500",
                }
              }
            >
              LOG IN
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "height": 150,
              "marginBottom": 50,
              "marginTop": 75,
              "width": 300,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Welcome, Jane Doe",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 18,
              "marginBottom": 30,
            },
          },
          "ref": null,
          "rendered": "Welcome, Jane Doe",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TextInput
              allowFontScaling={true}
              placeholder="Password"
              secureTextEntry={true}
              style={
                Object {
                  "height": 30,
                  "width": 300,
                }
              }
              underlineColorAndroid="rgb(0,163,158)"
            />,
            "style": Object {
              "borderBottomWidth": 1,
              "borderColor": "rgb(0,163,158)",
              "marginBottom": 20,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "placeholder": "Password",
              "secureTextEntry": true,
              "style": Object {
                "height": 30,
                "width": 300,
              },
              "underlineColorAndroid": "rgb(0,163,158)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#FFF",
                  "fontSize": 14,
                  "fontWeight": "500",
                }
              }
            >
              LOG IN
            </Text>,
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "backgroundColor": "rgb(0,163,158)",
              "borderRadius": 2,
              "elevation": 2,
              "height": 36,
              "marginBottom": 8,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "shadowColor": "rgba(0, 0, 0, 0.54)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
              "width": 300,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "LOG IN",
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "color": "#FFF",
                "fontSize": 14,
                "fontWeight": "500",
              },
            },
            "ref": null,
            "rendered": "LOG IN",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
